File: códigos en JAVA/ejercicio1ape6.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */

/**
 *
 * @author USUARIO
 */
    import java.util.ArrayList; // Importación de la clase ArrayList
public class ejercicio1ape6 {
    public static void main(String[] args) {
        // 1. Declaración e instanciación del ArrayList tipado con String
        ArrayList<String> nombres = new ArrayList<>();

        // 2. Inserción de datos mediante el método add()
        nombres.add("Ana");
        nombres.add("Luis");
        nombres.add("Carlos");
        nombres.add("Marta");
        nombres.add("Pedro");

        // 3. Recorrido del ArrayList usando un bucle for clásico y el método size()
        System.out.println("--- LISTA DE ESTUDIANTES ---");
        for (int i = 0; i < nombres.size(); i++) {
            // Se utiliza get(i) para acceder al elemento en la posición i
            System.out.println("Estudiante [" + i + "]: " + nombres.get(i));
        }
    }
}

File: códigos en C++/ejercicio1ape6.cpp
#include <iostream>
#include <vector> // Librería necesaria para usar arreglos dinámicos en C++
#include <string>

using namespace std;

int main() {
    // 1. Declaración de la estructura dinámica (vector de strings)
    vector<string> nombres;

    // 2. Inserción de datos usando el método push_back()
    nombres.push_back("Ana");
    nombres.push_back("Luis");
    nombres.push_back("Carlos");
    nombres.push_back("Marta");
    nombres.push_back("Pedro");

    // 3. Recorrido del vector utilizando su tamaño dinámico (size)
    cout << "--- LISTA DE ESTUDIANTES ---" << endl;
    for (size_t i = 0; i < nombres.size(); i++) {
        cout << "Estudiante [" << i << "]: " << nombres[i] << endl;
    }

    return 0;
}
